Mathlib/CategoryTheory/Limits/Shapes/Pullback/PullbackCone.lean

```diff
@@ -512,4 +512,79 @@ def PushoutCocone.isoMk {F : WalkingSpan ⥤ C} (t : Cocone F) :
       · dsimp
         simp
 
+section Under
+variable {f : X ⟶ Y} {g : X ⟶ Z} {c : PushoutCocone f g}
+
+/-- The binary cofan in `Under X` corresponding to a pushout cocone from `X`. -/
+@[simp]
+def PushoutCocone.toBinaryCofan : PushoutCocone f g ⥤ BinaryCofan (Under.mk f) (.mk g) where
+  obj c := .mk (Under.homMk (U := .mk f) (V := .mk (f ≫ c.inl)) c.inl rfl)
+      (Under.homMk (U := .mk g) (V := .mk (f ≫ c.inl)) c.inr c.condition.symm)
+  map {c1 c2} a := {
+    hom := Under.homMk a.hom
+    w := by rintro (_|_) <;> aesop_cat
+  }
+
+/-- The pushout cocone from `X` corresponding to a binary cofan in `Under X`. -/
+@[simp]
+def binaryCofanUnder.toPushoutCocone : BinaryCofan (Under.mk f) (.mk g) ⥤ PushoutCocone f g where
+  obj c := .mk c.inl.right c.inr.right (c.inl.w.symm.trans c.inr.w)
+  map {c1 c2} a := {
+    hom := a.hom.right
+    w := by rintro (_|_|_) <;> simp [← Under.comp_right]
+  }
+
+/-- Pushout cocones from `X` are the same thing as binary cofans in `Under X`. -/
+@[simp]
+def pushoutCoconeEquivBinaryCofan : PushoutCocone f g ≌ BinaryCofan (Under.mk f) (.mk g) :=
+  .mk PushoutCocone.toBinaryCofan binaryCofanUnder.toPushoutCocone
```

Comment:
Please do not use `mk` and use the `where` constructor instead (we should really add a warning at that def to tell people not to unless absolutely necessary).